packages/cli-kit/src/public/node/ui.tsx

```diff
@@ -267,13 +273,14 @@ export interface RenderConfirmationPromptOptions extends Pick<SelectPromptProps<
  *    Press ↑↓ arrows to select, enter or a shortcut to confirm
  *
  */
-export function renderConfirmationPrompt({
+export async function renderConfirmationPrompt({
   message,
   infoTable,
   confirmationMessage = 'Yes, confirm',
   cancellationMessage = 'No, cancel',
   renderOptions,
   defaultValue = true,
+  abortController,
```

Comment:
What you should be passing instead is the [signal](https://developer.mozilla.org/en-US/docs/Web/API/AbortSignal), the controller is only needed in the caller context to send an abort signal through.

Reply:
I was debating this. The reason I chose to pass the controller is that `ConcurrentOutput` already accepts a controller as a param so I wanted to keep these functions as similar as possible. We also don't know if in the future we'll need to call abort inside the components so I wanted to future proof this API so that we don't have to change it.